controllers/kserve_inferenceservice_controller_authconfig_test.go

```diff
@@ -0,0 +1,222 @@
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["anonymous-access"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+
+				Expect(enableAuth(isvc)).To(Succeed())
+				Eventually(func(g Gomega) {
+					ac := &authorinov1beta2.AuthConfig{}
+					g.Expect(ac.Spec.Authorization["kubernetes-user"]).NotTo(BeNil())
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+			})
+		})
+
+		Context("auth enabled", func() {
+			BeforeEach(func() {
+				isvc = createISVCWithAuth(namespace.Name)
+			})
+
+			It("should create user defined auth config", func() {
+				Expect(updateISVCStatus(isvc)).To(Succeed())
+
+				Eventually(func(g Gomega) {
+					ac := &authorinov1beta2.AuthConfig{}
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["kubernetes-user"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+			})
+
+			It("should update to anonymous on disable", func() {
+				Expect(updateISVCStatus(isvc)).To(Succeed())
+
+				Eventually(func(g Gomega) {
+					ac := &authorinov1beta2.AuthConfig{}
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["kubernetes-user"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+
+				Expect(disableAuth(isvc)).To(Succeed())
+				Eventually(func(g Gomega) {
+					ac := &authorinov1beta2.AuthConfig{}
+					g.Expect(getAuthConfig(namespace.Name, isvc.Name, ac)).To(Succeed())
+					g.Expect(ac.Spec.Authorization["anonymous-access"]).NotTo(BeNil())
+				}).
+					WithTimeout(timeout).
+					WithPolling(interval).
+					Should(Succeed())
+			})
+		})
+	})
+})
+
+func getAuthConfig(namespace, name string, ac *authorinov1beta2.AuthConfig) error {
+	return cli.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, ac)
+}
+
+func createISVCMissingStatus(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := &kservev1beta1.InferenceService{}
+	err := convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+	Expect(err).NotTo(HaveOccurred())
+	inferenceService.Namespace = namespace
+	Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+	return inferenceService
+}
+
+func createISVCWithAuth(namespace string) *kservev1beta1.InferenceService {
+	inferenceService := createBasicISVC(namespace)
+	inferenceService.Annotations["enable-auth"] = "true"
+	Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+	// HMM... second routine to do updates?
+	url, _ := apis.ParseURL("http://iscv-" + namespace + "ns.apps.openshift.ai")
+	inferenceService.Status = kservev1beta1.InferenceServiceStatus{
+		URL: url,
+	}
+
```

Comment:
This chunk of code seems to be safe to remove, given `updateISVCStatus` is called on the relevant units.